frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -82,50 +91,77 @@ function ManagePublicKeySwitchDisplay() {
         }
     };
 
+    const showPublicKeyConfigured =
+        sender.length > 0 &&
+        hasPublicKey &&
+        !uploadNewPublicKey &&
+        !fileSubmitted;
+    const showUploadMsg =
+        (sender.length > 0 && !fileSubmitted && !hasPublicKey) ||
+        (!uploadNewPublicKey && sender.length === 0);
+    const showUpload =
+        (sender.length > 0 && !fileSubmitted && !hasPublicKey) ||
+        uploadNewPublicKey;
+    const showUploadError = fileSubmitted && !isUploading && !isSuccess;
+
     return (
-        <>
-            {/*Waiting on backend to support this
+        <GridContainer className="manage-public-key padding-bottom-5 tablet:padding-top-6">
+            {!isUploading && (
+                <h1 className="margin-top-0 margin-bottom-3">
+                    Manage public key
+                </h1>
+            )}
+            {showUploadMsg && (
+                <>
+                    <p className="font-sans-md">
+                        Send your public key to begin the REST API
+                        authentication process.
+                    </p>
+                    <SiteAlert variant="info" showIcon={false}>
+                        <Icon.Lightbulb />
+                        <span className="padding-left-1">
+                            If you need more information on generating your
+                            public key, reference page 7 in the{" "}
+                            <USLink href="/resources/programmers-guide">
+                                API Programmer’s Guide.
+                            </USLink>
+                        </span>
+                    </SiteAlert>
+                </>
+            )}
             {sender.length === 0 && (
                 <ManagePublicKeyChooseSender
-                    onSenderSelect={(selectedSender: string) =>
-                        setSender(selectedSender)
-                    }
+                    onSenderSelect={handleSenderSelect}
                 />
             )}
-            {sender && !fileSubmitted && (*/}
-            {!fileSubmitted && (
+            {showPublicKeyConfigured && (
+                <ManagePublicKeyConfigured
+                    onUploadNewPublicKey={() => setUploadNewPublicKey(true)}
+                />
+            )}
+            {uploadNewPublicKey && (
+                <p className="font-sans-md">
+                    Your public key is already configured.
+                </p>
+            )}
+            {showUpload && (
                 <ManagePublicKeyUpload
                     onPublicKeySubmit={handlePublicKeySubmit}
                     onFileChange={handleFileChange}
+                    onBack={() => setSender("")}
+                    onFetchPublicKey={setHasPublicKey}
+                    showBack={showBack}
```

Comment:
Rather than having a `showBack` AND an `onBack`, could we just pass a function or null for `onBack`?  Then we could just check for that in the component:

```tsx
{ onBack && (
    <Button type="button" onClick={ onBack }>Back</Button>
) }
```